app/models/service_template_transformation_plan_request.rb

```diff
@@ -15,6 +15,16 @@ def source_vms
     vm_resources.where(:status => [ServiceResource::STATUS_QUEUED, ServiceResource::STATUS_FAILED]).pluck(:resource_id)
   end
 
+  def validate_conversion_hosts
+    transformation_mapping.transformation_mapping_items.each do |item|
+      next unless %w(EmsCluster CloudTenant).include?(item.source_type)
+      ems = item.destination_type.constantize.find(item.destination_id).ext_management_system
```

Comment:
This can just be `item.destination.ext_management_system`, `:destination` is polymorphic so rails basically does `_type.find(_id)` for you